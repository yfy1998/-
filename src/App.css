.App {
  text-align: center;
}

.App-label{
  text-align: right;
  width: 120px;
  height: 25px;
  margin-right: 8px;
  font-size: 20px;
  display: inline-block;
}

.App-input{
  margin-top: 20px;
  width: 300px;
  height: 30px;
  border-radius: 5px;
  font-size: 15px;
}

.App-button{
  width: 200px;
  height: 30px;
  border-radius: 5px;
  background-color: #61dafb;
  font-size: 20px;
  margin-top: 15px;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
  text-decoration: none;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
