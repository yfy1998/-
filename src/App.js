import React, { Component,Fragment } from 'react';
import Luggage from './Luggage'
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state= {
      flight: "in",
      area: "A",
      type: "a",
      highestprice: "",
      count: "",
      list: [],
      cost: "",
      touristtype: "a"
    }
  this.handleChange=this.handleChange.bind(this);
  this.handleLuggChange=this.handleLuggChange.bind(this);
  this.handleClick=this.handleClick.bind(this);
  }

  render() {
    return (
      <div className="App">
           <a
            className="App-link"
            href="http://www.csair.com/cn/tourguide/luggage_service/carryon_luggage/free_luggage/"
            target="_blank"
            rel="noopener noreferrer"
            >行李托运规则</a>
            <br />
          <label className="App-label">航班</label>
          <select className="App-input" name="flight" value={ this.state.flight } onChange={ this.handleChange }>
            <option value="in">国内航班</option>
            <option value="out">国际航班</option>
          </select>
          <br />
          <label className="App-label">舱位</label>
          <select className="App-input" name="type" value={ this.state.type } onChange={ this.handleChange }>
            <option value="a">头等舱</option>
            <option value="b">公务舱</option>
            <option value="c">经济舱</option>           
            <option value="d">不占座婴儿</option>
          </select>
          <br />
          <label className="App-label">游客类型</label>
          <select className="App-input" name="touristtype" value={ this.state.touristtype } onChange={ this.handleChange }>
            <option value="a">普通</option>
            <option value="b">南航明珠金卡会员、天合联盟超级精英</option>
            <option value="c">南航明珠银卡会员、天合联盟精英</option>           
            <option value="d">留学生、劳务、海员</option>
          </select>
          <br />
          {
            this.getflight()
          }
          <br />
          <label className="App-label">行李件数</label>
          <input 
            className="App-input"
            name="count" 
            type="text" 
            value={this.state.count}
            onChange={this.handleChange} 
            />
            <br />
            {
              this.getheight()
            }
          <button className="App-button" onClick={this.handleClick}>计算费用</button>
          <p style={{"whiteSpace":"pre","fontSize":"20px","color":"red"}}>{this.state.cost}</p>
      </div>
    );
  }

  handleClick(){
    let cos=0;
    let cost="";
    let flag=true;
    const type=this.state.type;
    const count=Number(this.state.count);
    const list=this.state.list;
    const area=this.state.area;
    const touristtype=this.state.touristtype;
    if(count===0){
      alert("未添加行李！");
      return;
    } 
    if(this.state.flight==="in"){
      let totalweight=0;
      for(let i in list){
        if(list[i].weight>50){
          cost+="第"+(Number(i)+1)+"件行李重量超过50kg，无法托运\r\n";
          flag=false;
          continue;
        }
        if(list[i].long>60){
          cost+="第"+(Number(i)+1)+"件行李长度超过60cm，无法托运\r\n";
          flag=false;
          continue;
        }
        if(list[i].width>40){
          cost+="第"+(Number(i)+1)+"件行李宽度超过40cm，无法托运\r\n";
          flag=false;
          continue;
        }
        if(list[i].height>100){
          cost+="第"+(Number(i)+1)+"件行李高度超过100cm，无法托运\r\n";
          flag=false;
          continue;
        }
        totalweight+=list[i].weight;
      }
      if(type==="a"&&totalweight>40){
        cos=this.caculincost(totalweight,40);
      }
      if(type==="b"&&totalweight>30){
        cos=this.caculincost(totalweight,30);
      }
      if(type==="c"&&totalweight>20){
        cos=this.caculincost(totalweight,20);
      }
      if(type==="d"&&totalweight>10){
        cos=this.caculincost(totalweight,10);
      }
      if(flag)  cost="总费用￥"+cos;
      else  cost=cost+"除无法托运的，其余总费用￥"+cos;
    }
    else{
      if(type==="d"){
        if(count>1){
          alert("不占座婴儿只能携带一件行李");
          return;
        }
        for(let i in list){
          let size=list[i].long+list[i].width+list[i].height;
          if(list[i].weight>10){
            alert("不占座婴儿行李重量不能超过10kg！");
            return;
          }
          if(size>115){
            alert("不占座婴儿行李尺寸（长宽高总和）不能超过115cm！")
          }
        }
      }
      if(area==="A"){
        if(type==="a"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,3,158,1000,2000,1000,3000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,4,158,1000,2000,1000,3000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,4,158,1000,2000,1000,3000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,4,158,1000,2000,1000,3000);
          }
        }
        if(type==="b"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,2,158,1000,2000,1000,3000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,3,158,1000,2000,1000,3000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,3,158,1000,2000,1000,3000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,3,158,1000,2000,1000,3000);
          }
        }
        if(type==="c"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,2,158,1000,2000,1000,3000,1000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,3,158,1000,2000,1000,3000,1000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,3,158,1000,2000,1000,3000,1000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,3,158,1000,2000,1000,3000,1000);
          }
        }
      }
      if(area==="B"){
        if(type==="a"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,3,158,450,1300,1000,3000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,4,158,450,1300,1000,3000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,4,158,450,1300,1000,3000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,4,158,450,1300,1000,3000);
          }
        }
        if(type==="b"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,2,158,450,1300,1000,3000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,3,158,450,1300,1000,3000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,3,158,450,1300,1000,3000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,3,158,450,1300,1000,3000);
          }
        }
        if(type==="c"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,1,158,450,1300,1000,3000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,2,158,450,1300,1000,3000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,2,158,450,1300,1000,3000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,2,158,450,1300,1000,3000);
          }
        }
      }
      if(area==="C"){
        if(type==="a"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,3,158,1000,2000,1000,3000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,4,158,1000,2000,1000,3000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,4,158,1000,2000,1000,3000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,4,158,1000,2000,1000,3000);
          }
        }
        if(type==="b"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,2,158,1000,2000,1000,3000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,3,158,1000,2000,1000,3000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,3,158,1000,2000,1000,3000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,3,158,1000,2000,1000,3000);
          }
        }
        if(type==="c"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,2,158,1000,2000,1000,3000,2000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,3,158,1000,2000,1000,3000,2000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,3,158,1000,2000,1000,3000,2000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,3,158,1000,2000,1000,3000,2000);
          }
        }
      }
      if(area==="D"){
        if(type==="a"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,3,158,450,1300,1000,3000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,4,158,450,1300,1000,3000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,4,158,450,1300,1000,3000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,4,158,450,1300,1000,3000);
          }
        }
        if(type==="b"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,3,158,450,1300,1000,3000,1000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,4,158,450,1300,1000,3000,1000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,4,158,450,1300,1000,3000,1000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,4,158,450,1300,1000,3000,1000);
          }
        }
        if(type==="c"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,2,158,450,1300,1000,3000,1000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,3,158,450,1300,1000,3000,1000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,3,158,450,1300,1000,3000,1000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,3,158,450,1300,1000,3000,1000);
          }
        }
      }
      if(area==="E"){
        if(type==="a"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,3,158,450,1300,1000,3000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,4,158,450,1300,1000,3000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,4,158,450,1300,1000,3000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,4,158,450,1300,1000,3000);
          }
        }
        if(type==="b"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,2,158,450,1300,1000,3000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,3,158,450,1300,1000,3000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,3,158,450,1300,1000,3000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,3,158,450,1300,1000,3000);
          }
        }
        if(type==="c"){
          if(touristtype==="a"){
            cost=this.caculoutcost(0,2,158,450,1300,1000,3000,1000);
          }
          if(touristtype==="b"){
            cost=this.caculoutcost(20,3,158,450,1300,1000,3000,1000);
          }
          if(touristtype==="c"){
            cost=this.caculoutcost(10,3,158,450,1300,1000,3000,1000);
          }
          if(touristtype==="d"){
            cost=this.caculoutcost(0,3,158,450,1300,1000,3000,1000);
          }
        }
      }
    }
    this.setState(() => ({
      cost: cost
    }));
  }

  caculoutcost(freeweight,freecount,freesize,cost1,cost2,cost3,cost4,cost5){
    const ncost5=cost5 || 0;
    const count=Number(this.state.count);
    const list=[...this.state.list];
    let res={
      cost: "",
      cos: 0
    };
    let arr=[];
    for(let i in list){
      if(list[i].weight>45){
        res.cost+="第"+(Number(i)+1)+"件行李重量超过45kg，无法托运\r\n";
        arr[i]=1;
        continue;
      }
      let size=list[i].long+list[i].width+list[i].height;
      if(size>300){
        res.cost+="第"+(Number(i)+1)+"件行李长宽高总和超过300cm，无法托运\r\n";
        arr[i]=1;
        continue;
      }
    }
    let faildcount=0;
    // eslint-disable-next-line
    for(let i in arr){
      faildcount++;
    }
    const leftcount=count-faildcount;
    if(faildcount>0) res.cost+="除无法托运的行李，其他可托运行李"+leftcount+"件\r\n";
    if(leftcount>freecount){
      const exceedcount=leftcount-freecount;
      const newcost=cost1+(exceedcount-1)*cost2;
      res.cost+="超件"+exceedcount+"件，费用￥"+newcost+"\r\n";
      res.cos+=newcost;
    }
    for(let i in list){
      if(arr[i]!==1){
        let size=list[i].long+list[i].width+list[i].height;
        if(size>freesize){
          res.cost+="第"+(Number(i)+1)+"件行李长宽高总和在159-300cm之间，超尺寸费用￥"+cost3+"\r\n";
          res.cos+=cost3;
        }
        if(list[i].weight>32+freeweight){
          res.cost+="第"+(Number(i)+1)+"件行李超重，超重费用￥"+cost4+"\r\n";
          res.cos+=cost4;         
        }
        if(ncost5!==0){
          if(list[i].weight>23+freeweight&&list[i]<=32+freeweight){
            res.cost+="第"+(Number(i)+1)+"件行李超重，超重费用￥"+ncost5+"\r\n";
            res.cos+=ncost5;             
          }
        }
      }
    }
    const cost=res.cost+"总费用￥"+res.cos;
    return cost;
  }

  caculincost(totalweight,num){
    const highestprice=this.state.highestprice;
    const touristtype=this.state.touristtype;
    let res=0;
    if(touristtype==="a"||touristtype==="d"){
      res=(totalweight-num)*highestprice*0.015;
    }
    if(touristtype==="b"){
      res=(totalweight-num-20)*highestprice*0.015;
    }
    if(touristtype==="c"){
      res=(totalweight-num-10)*highestprice*0.015;
    }
    if(res<0) res=0;
    return res;
  }

  getheight(){
    return this.state.list.map((item,index) => {
      return (
        <Luggage 
          key={index}
          index={index} 
          item={item}  
          LugChange={this.handleLuggChange}
          />
      );
    })
  }

  handleLuggChange(index,name,value){
    const list=[...this.state.list];
    //const list = JSON.parse(JSON.stringify(this.state.list));
    //list[index][name]=value;
    list[index] = {
      ...list[index],
      [name]: value
    }
    this.setState( () => ({
      list: list
    } ));
    //console.log(this.state.list);
  }

  getflight(){
    if(this.state.flight==="in"){
      return(
        <Fragment>
          <label className="App-label">单日最高票价</label>
          <input 
            className="App-input"
            name="highestprice" 
            type="text" 
            value={this.state.highestprice}
            onChange={ this.handleChange }
            />
        </Fragment>
      );
    }
    else return(
      <Fragment>
          <label className="App-label">区域</label>        
            <select 
            className="App-input"
            name="area" 
            value={ this.state.area } 
            onChange={ this.handleChange }
            >
              <option value="A">
                涉及日本、美洲、澳新、俄罗斯、迪拜的航程，以及新加坡始发与中国大陆间的航程
              </option>
              <option value="B">
                涉及中西亚的航程
              </option>
              <option value="C">
                涉及内罗毕及土耳其的航程
              </option>
              <option value="D">
                除涉及日本、美洲、澳新、俄罗斯、迪拜、内罗毕、中西亚及土耳其以外的国际/地区航程
              </option>
              <option value="E">
                涉及韩国始发与中国间的航程
              </option>
          </select>
      </Fragment>
    );
  }

  handleChange(event){
    const value=event.target.value;
    const name=event.target.name;
    //console.log(Number(value));
    if(name==="count") {
      if(isNaN(Number(value)) || value[value.length-1]==="."){
        //alert("行李件数只能输入自然数！");
        return;
      }
      const list=[];
      list.length=value;
      list.fill({});
      this.setState({
        list: list
      });
    }
    if(name==="highestprice"){
      if(isNaN(Number(value))){
        //alert("票价只能输入非负数！");
        return;
      }
    }
    this.setState(() => ({
      [name]: value
    }));
  }
}

export default App;
