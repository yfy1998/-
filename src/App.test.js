import React from 'react';
import ReactDOM from 'react-dom';
import { render,mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

//jest.mock('./App');

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("App test",() => {
    let wrapper=mount(<App />);
    wrapper.setState(()=>({
      highestprice: 1000,
      count: 2,
      list: [
        {weight:50,long:50,width:30,height:100},
        {weight:20,long:50,width:30,height:100}
      ],
    }))

    //内部数据初始化测试
    it('before click',() => {
      expect(wrapper.state('cost')).toBe("");
      expect(wrapper.state('flight')).toBe("in");
      expect(wrapper.state('area')).toBe("A");
      expect(wrapper.state('type')).toBe("a");
      expect(wrapper.state('touristtype')).toBe("a");
    });

    it('after click ', () => {
      expect(wrapper.find('input').exists());
      wrapper.find('button').simulate('click');
      expect(wrapper.state('cost')).toBe("总费用￥450");
    });

    it('when click button,handleClick() should be called',() => {
      const spyFn = jest.spyOn(App.prototype, 'handleClick');
      wrapper.find('button').simulate('click');
      expect(spyFn).not.toHaveBeenCalled();
      spyFn.mockRestore();
    });

    it('when first input change 5,onChang() should be called', () => {
      wrapper.find('input').at(0).simulate('change', {
          target: {
            value: '5'
          }});
          expect(wrapper.state('highestprice')).toBe(1000);
    });

    it('when last input change,onChang() should be called', () => {
      wrapper.find('input').last().simulate('change', {
          target: {
            name: 'height',
            value: '5'
          }});
          expect(wrapper.state('list')[1].height).toBe(5);
    });
})

it('basic use', () => {
  const wrapper = render(<App />);
  expect(toJson(wrapper)).toMatchSnapshot();
})

describe("App test1",() => {
  let wrapper=mount(<App />);
  wrapper.setState(()=>({
    highestprice: 1000,
    count: 2,
    list: [
      {weight:50,long:50,width:30,height:100},
      {weight:20,long:50,width:30,height:100}
    ],
    touristtype: 'b',
    type: 'b'
  }))

  it('after click 1', () => {
    expect(wrapper.find('input').exists());
    wrapper.find('button').simulate('click');
    expect(wrapper.state('cost')).toBe("总费用￥300");
  });
})

describe("App test2",() => {
  let wrapper=mount(<App />);
  wrapper.setState(()=>({
    highestprice: 1000,
    count: 1,
    list: [
      {weight:20,long:50,width:30,height:100}
    ],
    type: 'd'
  }))

  it('after click 2', () => {
    expect(wrapper.find('input').exists());
    wrapper.find('button').simulate('click');
    expect(wrapper.state('cost')).toBe("总费用￥150");
  });
})

describe("App test3",() => {
  let wrapper=mount(<App />);
  wrapper.setState(()=>({
    highestprice: 1000,
    count: 1,
    list: [
      {weight:50,long:50,width:30,height:100}
    ],
    touristtype: 'c',
    type: 'c'
  }))

  it('after click 3', () => {
    expect(wrapper.find('input').exists());
    wrapper.find('button').simulate('click');
    expect(wrapper.state('cost')).toBe("总费用￥300");
  });
})

describe("App test4",() => {
  let wrapper=mount(<App />);
  wrapper.setState(()=>({
    highestprice: 1000,
    count: 1,
    list: [
      {weight:50,long:50,width:30,height:100}
    ],
    touristtype: 'c',
    type: 'b',
    flight: 'in'
  }))

  it('after click 4', () => {
    expect(wrapper.find('input').exists());
    wrapper.find('button').simulate('click');
    expect(wrapper.state('cost')).toBe("总费用￥150");
  });
})

describe("App test5",() => {
  let wrapper=mount(<App />);
  wrapper.setState(()=>({
    highestprice: 1000,
    count: 2,
    list: [
      {weight:40,long:50,width:30,height:100},
      {weight:40,long:50,width:30,height:100}
    ],
    touristtype: 'c',
    type: 'a',
    flight: 'out'
  }))

  it('after click 5', () => {
    expect(wrapper.find('input').exists());
    wrapper.find('button').simulate('click');
    //expect(wrapper.state('cost')).toBe("总第1件行李长宽高总和在159-300cm之间，超尺寸费用￥1000第2件行李长宽高总和在159-300cm之间，超尺寸费用￥1000总费用￥2000");
  });
})


