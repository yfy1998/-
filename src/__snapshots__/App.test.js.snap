// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic use 1`] = `
<div
  class="App"
>
  <a
    class="App-link"
    href="http://www.csair.com/cn/tourguide/luggage_service/carryon_luggage/free_luggage/"
    rel="noopener noreferrer"
    target="_blank"
  >
    行李托运规则
  </a>
  <br />
  <label
    class="App-label"
  >
    航班
  </label>
  <select
    class="App-input"
    name="flight"
  >
    <option
      selected=""
      value="in"
    >
      国内航班
    </option>
    <option
      value="out"
    >
      国际航班
    </option>
  </select>
  <br />
  <label
    class="App-label"
  >
    舱位
  </label>
  <select
    class="App-input"
    name="type"
  >
    <option
      selected=""
      value="a"
    >
      头等舱
    </option>
    <option
      value="b"
    >
      公务舱
    </option>
    <option
      value="c"
    >
      经济舱
    </option>
    <option
      value="d"
    >
      不占座婴儿
    </option>
  </select>
  <br />
  <label
    class="App-label"
  >
    游客类型
  </label>
  <select
    class="App-input"
    name="touristtype"
  >
    <option
      selected=""
      value="a"
    >
      普通
    </option>
    <option
      value="b"
    >
      南航明珠金卡会员、天合联盟超级精英
    </option>
    <option
      value="c"
    >
      南航明珠银卡会员、天合联盟精英
    </option>
    <option
      value="d"
    >
      留学生、劳务、海员
    </option>
  </select>
  <br />
  <label
    class="App-label"
  >
    单日最高票价
  </label>
  <input
    class="App-input"
    name="highestprice"
    type="text"
    value=""
  />
  <br />
  <label
    class="App-label"
  >
    行李件数
  </label>
  <input
    class="App-input"
    name="count"
    type="text"
    value=""
  />
  <br />
  <button
    class="App-button"
  >
    计算费用
  </button>
  <p
    style="white-space:pre;font-size:20px;color:red"
  />
</div>
`;
